### ISSUE TITLE
[iOS] Build failed - OptimoveReactNative-Swift.h file not found

### ISSUE THREAD

**Issue by antoxi4** [optimove-tech/Optimove-SDK-React-Native#9]
When i've tried build app i've got an error OptimoveReactNative-Swift.h file not found in OptimoveReactNative.m

XCode 14.1
App Configuration:
```
react-native - 0.70.6
@optimove-inc/react-native - 1.1.1
```

Podfiles Configuration:
```
target 'NotificationServiceExtension' do
  use_frameworks!
  pod 'OptimoveNotificationServiceExtension', '~> 5.0'
end
```

Following [Apple Documentation](https://developer.apple.com/documentation/swift/importing-swift-into-objective-c#Import-Code-Within-an-App-Target) using different Pods variants (use_frameworks: true | false) it requires different ways to import swift implementations into Objective C

There is workaround how to manage different imports in single file

```
#if __has_include(<OptimoveReactNative/OptimoveReactNative-Swift.h>) // with use_framework flag
#import <OptimoveReactNative/OptimoveReactNative-Swift.h>

#elif __has_include("OptimoveReactNative-Swift.h") // without use_framework flag
#import "OptimoveReactNative-Swift.h"

#endif
```

**Comment by cgwyllie** [comment]
Hi @antoxi4,  thanks for reporting, can you please indicate your Xcode version?

**Comment by antoxi4** [comment]
> Hi @antoxi4, thanks for reporting, can you please indicate your Xcode version?

Yep, added in first message.

**Comment by cgwyllie** [comment]
Could you also show your full Podfile for the app target as well?

Thanks!

**Comment by antoxi4** [comment]
> Hi @antoxi4 , whilst we're looking at making this compatible with both frameworks & static libraries, is there any preference for one or the other in your proejct?
> 
> For example, by default RN pod projects do not set use_frameworks to true, was there a particular reason to switch for you?
> 
> Thanks

Yeah, the reason for use_frameworks is rn-firebase. This package can works only with use_frameworks flag.
Regarding question about full Podfile.


```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

$RNFirebaseAsStaticFramework = true # added manualy

platform :ios, '12.4'
install! 'cocoapods', :deterministic_uuids => false

target 'BeyondBody' do
  use_frameworks! :linkage => :static # added manualy

  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => true,
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    # :flipper_configuration => FlipperConfiguration.enabled,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  post_install do |installer|
    react_native_post_install(
      installer,
      # Set `mac_catalyst_enabled` to `true` in order to apply patches
      # necessary for Mac Catalyst builds
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end

target 'NotificationServiceExtension' do # Your new extension target
  use_frameworks!
  pod 'OptimoveNotificationServiceExtension', '~> 5.0'
end
```

**Comment by cgwyllie** [comment]
Hi @antoxi4 , thanks for the further information, very useful context.

The fix for this issue will be included in our 2.0.0 release which we are in the process of publishing now.

Feel free to re-open if you have any further questions around this!
